prov/gni/src/gnix_rma.c

```diff
@@ -740,6 +795,152 @@ int _gnix_rma_post_rdma_chain_req(void *data)
 	return FI_SUCCESS;
 }
 
+static void __gnix_rma_more_fill_pd(struct gnix_fab_req *req,
+				    struct gnix_tx_descriptor *txd)
+{
+	gni_ct_put_post_descriptor_t *more_put;
+	gni_ct_get_post_descriptor_t *more_get;
+	gni_mem_handle_t mdh;
+	struct gnix_fab_req *more_req;
+	struct slist_entry *current;
+	int entries = 0, idx = 0;
+
+	current = req->rma.sle.next; /* First sub descriptor slist entry */
```

Comment:
The reason that we chose to not use an slist iterator here, is because the slist is on the EP, and needs to be cleared out periodically. 
If a message fails to transmit, and tries to re-transmit, it will return to this work_fn. The contents of the slist on the EP by that point may be the next chain of FI_MORE requests. 

When the retry of the top level request happens, we can still find all the sub-requests that need to be chained by using this method. 
Does that seem reasonable/make sense?